### PR TITLE
feat: add font size selection to the preferences [ACC-302]

### PR DESCRIPTION
**Pull request by V-Gira**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-302" title="ACC-302" target="_blank">StoryACC-302</a>  Font Scaling in Preferences (Part of BITV 11.7)
  </td></table>
  <br />

### CODE REVIEW

File: src/script/hooks/useRootFontSize.ts [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+function setFontSizeToRoot(currentRootFontSize: RootFontSize) {
+  const root = document.documentElement;
+  root.style.fontSize = currentRootFontSize.toString();
+}
+
+export const useRootFontSize = (
+  rootFontSize: RootFontSize = RootFontSize.M,
+): [RootFontSize, React.Dispatch<React.SetStateAction<RootFontSize>>] => {
+  const [storedRootFontSize, setStoredRootFontSize] = useLocalStorage<RootFontSize>('root-font-size', rootFontSize);
```
Comment (przemvs) [inline src/script/hooks/useRootFontSize.ts]: Move 'root-font-size' key to const. You can export it in this hook. e.g.
export const ROOT_FONT_KEY = '...';

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx [hunk 2]
```diff
@@ -161,7 +183,15 @@ const OptionPreferences: React.FC<OptionPreferencesProps> = ({
 
           <hr className="preferences-separator" />
 
-          <PreferencesSection title={t('preferencesOptionsPopular')}>
+          <PreferencesSection title={t('preferencesOptionsAppearance')}>
+            <div css={{marginBottom: '1.5rem', width: '100%'}}>
```
Comment (arjita-mitra) [inline src/script/page/MainContent/panels/preferences/OptionPreferences.tsx]: why are we using inline CSS?

File: src/script/page/RightSidebar/GroupParticipantUser/GroupParticipantUser.tsx [hunk 3]
```diff
@@ -201,7 +201,7 @@ const GroupParticipantUser: FC<GroupParticipantUserProps> = ({
               </div>
             </div>
 
-            <div className="panel__action-item panel__info-text panel__item-offset" tabIndex={0}>
+            <div className="panel__info-text panel__item-offset" css={{padding: '16px'}} tabIndex={0}>
```
Comment (arjita-mitra) [inline src/script/page/RightSidebar/GroupParticipantUser/GroupParticipantUser.tsx:204]: why are we using inline CSS?

File: src/style/foundation/video-calling.less [hunk 4]
```diff
@@ -175,7 +175,7 @@
       left: 50%;
       margin: 0;
       color: var(--main-color);
-      font-size: @font-size-xsmall;
+      font-size: 11px;
```
Comment (arjita-mitra) [inline src/style/foundation/video-calling.less:178]: shouldn't we use em here?

File: src/script/hooks/useRootFontSize.ts [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+export enum RootFontSize {
+  XXS = '10px',
+  XS = '12px',
+  S = '14px',
+  M = '16px',
+  L = '18px',
+  XL = '20px',
+  XXL = '24px',
+  // XXXL = '30px',
+  // XXXXL = '36px',
```
Comment (arjita-mitra) [inline src/script/hooks/useRootFontSize.ts]: remove commented code

**Review by arjita-mitra — CHANGES_REQUESTED** [review]

File: src/style/foundation/video-calling.less (hunk 4 above)
Comment (V-Gira) [reply src/style/foundation/video-calling.less:178]: We decided to keep a px value for some elements for now, mainly the call ui and the sidebar button

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx (hunk 2 above)
Comment (V-Gira) [reply src/script/page/MainContent/panels/preferences/OptionPreferences.tsx]: I should probably raise that on the web chat, I'm under the impression we're moving toward inline CSS for oneoff rules like this

File: src/script/page/AppMain.tsx [hunk 6]
```diff
@@ -74,6 +75,19 @@ const AppMain: FC<AppMainProps> = ({
...
+  useEffect(() => {
+    PrimaryModal.init();
+    showInitialModal(userAvailability);
+    // userAvailability not needed for dependency
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useLayoutEffect(() => {
+    initializeApp();
+  }, []);
```
Comment (atomrc) [inline src/script/page/AppMain.tsx]: I think this look a little weird, we consume `initializeApp` before it was defined. 
Let's re-move this back to where is was initially

File: src/script/page/MainContent/panels/preferences/OptionPreferences.tsx (hunk 2 above)
Comment (atomrc) [reply src/script/page/MainContent/panels/preferences/OptionPreferences.tsx]: I'm personally, not a big fan of inline css. 
I'm not formally against it either, but when possible it's be nice to extract those into a `Component.styles.ts`

**Pull request merged**